### ISSUE TITLE
QGIS crashes after a few seconds each time I open it.

### ISSUE THREAD

**Issue by Vertigrator** [qgis/QGIS#41934]
## User Feedback

I get this crash when opening the client - after about 5 seconds. I was on version 3.6. So I upgraded to version 3.18 and it is still crashing.

## Report Details

**Crash ID**: [67d946f7e16887bfd939dfe454d37ea8d66bc49c](https://github.com/qgis/QGIS/search?q=67d946f7e16887bfd939dfe454d37ea8d66bc49c&type=Issues)

**Stack Trace**
<pre>
QSslKey::operator!= :
QSslKey::operator!= :
QObject::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:471
QCoreApplication::notifyInternal2 :
QCoreApplicationPrivate::sendPostedEvents :
QEventDispatcherWin32::processEvents :
UserCallWinProcCheckWow :
DispatchMessageWorker :
QEventDispatcherWin32::processEvents :
QEventLoop::exec :
QThread::exec :
QThread::start :
BaseThreadInitThunk :
RtlUserThreadStart :
</pre>

**QGIS Info**
QGIS Version: 3.18.0-Z�rich
QGIS code revision: bdef9fb328
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.1.4
Running against GDAL: 3.1.4

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.18363

**Comment by gioman** [comment]
@Vertigrator try with a new QGIS profile.

**Comment by gioman** [comment]
> when opening the client

@Vertigrator also you may want to explain what "opening the client" means.

**Comment by roya0045** [comment]
@gioman duplicate of #37121 maybe it should be reopened and close this one.

**Comment by Vertigrator** [comment]
> 
> 
> > when opening the client
> 
> @Vertigrator also you may want to explain what "opening the client" means.

I mean simply opening the application.

**Comment by Vertigrator** [comment]
> 
> 
> @Vertigrator try with a new QGIS profile.

I'm not sure what creating a new QGIS profile means. I completely uninstalled all versions of QGIS and then installed 3.18. Is that creating a new profile?

**Comment by Vertigrator** [comment]
I have now installed it on a machine it has never been installed on before, using v3.16 and the same thing is happening. The screen is frozen as soon as you launch the application and after 5 seconds or so it comes up with the crash dialog screen.

**Comment by Vertigrator** [comment]
I have found the problem now. It appears to be something that our network admins have done with the firewall/proxy. By choosing a different proxy to go through with less restrictions it opened properly and the difference was it now loads the News feed. So this must have been getting blocked before and eventually causing it to lockup and crash.

**Comment by Vertigrator** [comment]
A fix for this was to un-tick the option 'Show QGIS news feed on welcome page' in the options\general tab. Then I was able to go back to the default proxy settings for my organisation and it wasn't trying to grab the news so it is no longer crashing.

**Comment by gioman** [comment]
> A fix for this was to un-tick the option 'Show QGIS news feed on welcome page' in the options\general tab.

@Vertigrator so to resume:

There is a scenario that when a computer is behind a proxy configured in some specific way and QGIS has the  'Show QGIS news feed on welcome page' active, then QGIS crashes on start, is this correct?

**Comment by Vertigrator** [comment]
> 
> 
> > A fix for this was to un-tick the option 'Show QGIS news feed on welcome page' in the options\general tab.
> 
> @Vertigrator so to resume:
> 
> There is a scenario that when a computer is behind a proxy configured in some specific way and QGIS has the 'Show QGIS news feed on welcome page' active, then QGIS crashes on start, is this correct?

Yes that is correct. On both PCs I was able to fix the issue by changing to a different proxy in 'internet options' control panel, then the program stopped crashing on launch. I was then able to go into settings and disable the new feed. Then when I swapped back to the standard work proxy server it was able to launch without crashing still, because it was no longer trying to fetch the news feed.

**Comment by gioman** [comment]
> by changing to a different proxy

@Vertigrator can you share the offending proxy configurations?

**Comment by Vertigrator** [comment]
I just talked to our network admin and they say that our internal proxy expects user credentials to be passed through whenever requesting things externally.
When I went to Settings\Options\Network tab, QGIS would lock up and crash. 
I changed back to the other proxy that doesn't require credentials. 
I was then able to get the Settings\Options\Network tab to load. I was then able to tick the checkbox for 'Use proxy for web access'. This was not ticked by default.
I was then able to switch back to our organisations proxy server that requires authentication. I can now bring in the news feed and I can now get to the Settings\Options\Network tab, without it locking up and crashing any more.
So I suggest there needs to be a try catch with a timeout or similar when trying to access the external world. Maybe give the user an error message explaining that QGIS couldn't connect and that you need to select the correct settings in the network tab. But of course I could only select that once I had stopped using our organisations proxy, because it was locked/frozen and then crashes. So if you have a try/catch with timeout you should be able to stop it getting locked up.
